### PR TITLE
tests: clean up logic for repositories in various test scenarios

### PR DESCRIPTION
**Pull request by KKoukiou**
* Always pull anaconda-core from COPR build from master branch, previously it was happening only if TEST_SCENARIO was not empty (fix for 082c3601a394b51b9b3204de82132ad070bd28b5)
* Add scenario for anaconda-webui PR

Document all options better.

### CODE REVIEW

File: test/build-rpms [hunk 1]
```diff
@@ -27,30 +27,41 @@ def build_rpms(srpm, image, verbose, quick):
...
-
-        # download cockpit rpms
-        # FIXME boot.iso on rawhide does not currently contain the new cockpit dependencies
+        # Pull cockpit dependencies from the image default compose
+        # unless we are testing a PR on cockpit-project/cockpit, then pull it from the PR COPR repo
+        if scenario and scenario.startswith("cockpit-pr-"):
+            cockpit_pr = scenario.split("-")[-1]
+            machine.execute(f"dnf copr enable -y packit/cockpit-project-cockpit-{cockpit_pr}", stdout=sys.stdout)
+
+        # Build anaconda-webui from SRPM unless we are testing a PR, then pull it from the PR COPT repo
```
Comment (martinpitt) [inline test/build-rpms:38]: I suspect you mean the opposite, "**if** we are testing a PR, otherwise pull it ..."? (this sentence implies it's an anaconda-webui PR) Because then you do want to build the rpms from the srpm that you just built. 

TBH the whole structure is a bit muddy: It seems the Makefile always builds an srpm from the checkout (PR for an anaconda-webui PR, main for a cockpit or anaconda PR), but the srpm gets ignored in the latter two cases, right? There could be a comment to point that out, that it is intended. Or better yet, shift all of this one level up, as for testing a cockpit or anaconda PR there shouldn't be any rpm build involved in the first place, so invoking a "build-rpms" script sounds weird.

Or perhaps rename "build-rpms" to "vm.install" or something, and move the whole setup here?

**Review by martinpitt — APPROVED** [review]
I didn't see anything obviously wrong, but TBH it's hard to follow what happens here. So if you are in a hurry and want to get unblocked, go ahead.

File: test/vm.install [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+BOTS_DIR = os.path.realpath(f'{__file__}/../../bots')
+sys.path.append(BOTS_DIR)
+
+missing_packages = "cockpit-ws cockpit-bridge cockpit-storaged fedora-logos"
+# Install missing firefox dependencies.
+# Resolving all dependencies with dnf download is possible,
+# but it packs to many packages to updates.img
+missing_packages_incl_deps = "firefox"
+
+from machine.machine_core import machine_virtual  # NOQA: imported through parent.py
```
Comment (martinpitt) [inline test/vm.install:21]: FTR, there is no `parent.py` any more.

**Comment by martinpitt** [comment]
[QR code is missing](https://cockpit-logs.us-east-1.linodeobjects.com/pull-75-20231204-140648-dcc637c8-fedora-rawhide-boot/log.html#11_), but this feels like a flake. Retrying.

File: test/vm.install [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                            f"{mock_opts} --rebuild /var/tmp/build/SRPMS/*.src.rpm'", timeout=1800)
+        else:
+            anaconda_webui_pr = scenario.split("-")[-1]
+            machine.execute(f"dnf copr enable -y packit/rhinstaller-anaconda-webui-{anaconda_webui_pr}", stdout=sys.stdout)
+            packages_to_download += f" anaconda-webui"
+
+        # Pull anaconda-core from the COPR repo packit builds from master branch
+        # unless we are testing a PR on rhinstaller/anaconda, then pull it from the PR COPR repo
+        if not scenario or not scenario.startswith("anaconda-pr-"):
+            machine.execute(f"dnf copr enable -y @rhinstaller/Anaconda ", stdout=sys.stdout)
```
Comment (martinpitt) [inline test/vm.install:61]: Ah, very good, this adds the previously missing `-y`. That currently causes all tests on cockpit PRs to [fail like this](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19689-20231205-050231-351207a7-fedora-rawhide-boot-cockpit-pr-19689-rhinstaller-anaconda-webui/log.html).

**Review by martinpitt — APPROVED** [review]
LGTM now, thanks! Let's see if the retry is happy.

**Comment by KKoukiou** [comment]
> [QR code is missing](https://cockpit-logs.us-east-1.linodeobjects.com/pull-75-20231204-140648-dcc637c8-fedora-rawhide-boot/log.html#11_), but this feels like a flake. Retrying.

@martinpitt these were wrongly submitted pixel test reference from an unmerged PR ;) removed!

File: test/vm.install (hunk 2 above)
Comment (KKoukiou) [reply test/vm.install:21]: Thanks for heads up! Will clean it up in followup

File: test/vm.install (hunk 3 above)
Comment (KKoukiou) [reply test/vm.install:61]: This will fix it, Sorry!
